packages/langium-vscode/src/commands.ts

```diff
@@ -0,0 +1,92 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+import * as vscode from 'vscode';
+import { exec } from 'child_process';
+import * as path from 'path';
+import fs from 'fs-extra';
+
+export async function generateTypesHandler(outputChannel: vscode.OutputChannel, ...args: string[]): Promise<void> {
+    outputChannel.show(true);
+    const { execute, log, logError } = executor(outputChannel);
+    const filePath: vscode.Uri | undefined = (args && args.length > 1) ? vscode.Uri.parse(args[1]) : vscode.window.activeTextEditor?.document.uri;
+    if (!filePath) {
+        log('Langium file URI is not provided. Execution canceled.');
+        return;
+    }
+    let cliVersion;
+    try {
+        cliVersion = await execute('langium -V', true);
+        // TODO add version compatibility check
+        log('Using langium-cli v' + String(cliVersion).trim());
+    } catch (error) {
+        // not installed
+        const installCmd = 'npm install -g langium-cli';
```

Comment:
I wouldn't expect users to install the `langium-cli` globally. The extension should instead use the package that is in the `node_modules`. TypeScript for example works the same way.

Reply:
> The extension should instead use the package that is in the node_modules

I disagree here. Langium extension as part of the tooling should **not** use the project setup, but a compatible langium-cli version installed globally. Imagine you have a Langium project in v0.5.0 and Langium extension in v0.6.0, this will not work as the released v0.5.0 langium-cli lacks corresponding CLI command implementation. 

However, I would love to see how TypeScript manages that with `npm_modules` folder, could you point me to the corresponding repository? We could offer this as an option before starting generation.

Btw. we suggest users to install example cli generators globally, is that something we can improve/change?

Reply:
> Langium extension as part of the tooling should not use the project setup, but a compatible langium-cli version installed globally.

I strongly disagree. Depending on a globally installed `langium-cli` breaks reproducibility of builds. You should always add it to `devDependencies`, and any tooling should use the local version if available.